### PR TITLE
Migrate unit tests Part 7

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## Description

Provide a summary of the modification as well as the issue that has been resolved.

- Migrate remaining tests.

## Changes

<details>
<summary>Describe the changes you made</summary>

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

</details>

## Checklist

<details>
<summary>Ensure that you followed the following</summary>

- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas (If applicable)
- [ ] I have made corresponding changes to the documentation (If applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (If applicable)

</details>

### CODE REVIEW

File: tests/unit/deploy/test_inferencer.py [hunk 1]
```diff
@@ -14,7 +15,35 @@
...
-from tests.legacy.helpers.inference import MockImageLoader
+
+
+class _MockImageLoader:
+    """Create mock images for inference on CPU based on the specifics of the original torch test dataset.
+
+    Uses yield so as to avoid storing everything in the memory.
+
+    Args:
+        image_size (List[int]): Size of input image
```
Comment (samet-akcay) [inline tests/unit/deploy/test_inferencer.py:26]:
```suggestion
        image_size (list[int]): Size of input image
```

File: tests/unit/metrics/aupro/aupro_reference.py [hunk 2]
```diff
@@ -56,29 +62,31 @@ def compute_overlap(self, threshold):
...
         Area under the curve.
     """
-
     x = np.array(x)
     y = np.array(y)
     finite_mask = np.logical_and(np.isfinite(x), np.isfinite(y))
     if not finite_mask.all():
-        print(
-            "WARNING: Not all x and y values passed to trapezoid are finite. Will continue with only the finite values."
+        logger.warning(
```
Comment (samet-akcay) [inline tests/unit/metrics/aupro/aupro_reference.py:87]: does pre-commit like this? Thought it would want a message variable

**Review by samet-akcay — CHANGES_REQUESTED** [review]
nice one. Only had a single suggested change

File: tests/unit/metrics/aupro/aupro_reference.py (hunk 2 above)
Comment (ashwinvaidya17) [reply tests/unit/metrics/aupro/aupro_reference.py:87]: I didn't get any issue with pre-commit for this one. I think this is required only for Exceptions.

**Review by samet-akcay — APPROVED** [review]

**Pull request merged**